### ISSUE TITLE
The listItem must be allowed where $block is, not only in $root

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-list#34]
Current rules:

```js

		schema.registerItem( 'listItem', '$block' );
		schema.allow( {
			name: 'listItem',
			inside: '$root',
			attributes: [ 'type', 'indent' ]
		} );
		schema.requireAttributes( 'listItem', [ 'type', 'indent' ] );
```

The `inside: '$root'` piece is breaking the quote feature because at the same time list attributes are required, but are not in `$root > quote > listItem` case.

**Comment by Reinmar** [comment]
Uh... turns out that Schema doesn't support such case now ;/

I pushed t/34: https://github.com/ckeditor/ckeditor5-list/compare/t/34

Unfortunately, the second test fails. This explains why the `inside: '$root'` was there.

**Comment by Reinmar** [comment]
Fixed with ckeditor/ckeditor5-engine#532 and https://github.com/ckeditor/ckeditor5-list/commit/a2395fbf0c2bff622963fe6776769f770c6a269c.
